src/everett/component.py

```diff
@@ -128,13 +139,13 @@ def get_required_config(cls):
         """
         options = ConfigOptions()
         for cls in reversed(cls.__mro__):
-            try:
-                options.update(cls.required_config)
-            except AttributeError:
-                pass
+            if hasattr(cls, "required_config"):
+                options.update(getattr(cls, "required_config"))
         return options
 
-    def get_runtime_config(self, namespace=None):
+    def get_runtime_config(
+        self, namespace: Union[List[str], None] = None
+    ) -> Generator[Tuple[Union[List[str], None], str, Any, Option], None, None]:
```

Comment:
This is kind of gross looking. When we drop support for Python 3.6, we can switch to data classes and drop the tuple here for a dataclass.